### ISSUE TITLE
Vmware Import Problem

### ISSUE THREAD

**Issue by omurozlu** [apache/cloudstack#8712]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Vmware
~~~

##### CLOUDSTACK VERSION

~~~
4.19
~~~

##### SUMMARY

Storage is not seen while importing [ 4.18.1.0 Working.]

<img width="1356" alt="Screenshot 2024-02-27 at 22 08 22" src="https://github.com/apache/cloudstack/assets/41322804/f026653d-81ac-45f9-9e05-630d0b6bf933">

**Comment by sureshanaparti** [comment]
@omurozlu Is that storage added as primary storage in CloudStack, cluster-wide or zone-wide?

**Comment by omurozlu** [comment]
hello @sureshanaparti,

I tried it both ways. (zone + cluster)
<img width="2273" alt="Screenshot 2024-02-28 at 10 03 48" src="https://github.com/apache/cloudstack/assets/41322804/2ecad03f-da44-4e4d-9eef-79ef84325b09">

**Comment by sureshanaparti** [comment]
Hi @omurozlu I did quick code check, seems issue with datastore path comparison here

https://github.com/apache/cloudstack/blob/45d267ccbf2749c547cbbbac4a2cb1f3351dcaf2/server/src/main/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImpl.java#L554  (updated in https://github.com/apache/cloudstack/pull/8445)

**Comment by omurozlu** [comment]
hello @sureshanaparti,
im check api post process.
Do I need a manual process to do this?

<img width="478" alt="Screenshot 2024-02-28 at 10 44 45" src="https://github.com/apache/cloudstack/assets/41322804/cd54dd77-3509-4ab0-9195-455f819dac1b">

**Comment by sureshanaparti** [comment]
> hello @sureshanaparti, im check api post process. Do I need a manual process to do this?
> 
> <img alt="Screenshot 2024-02-28 at 10 44 45" width="478" src="https://private-user-images.githubusercontent.com/41322804/308463619-cd54dd77-3509-4ab0-9195-455f819dac1b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkxMTA4NDQsIm5iZiI6MTcwOTExMDU0NCwicGF0aCI6Ii80MTMyMjgwNC8zMDg0NjM2MTktY2Q1NGRkNzctMzUwOS00YWIwLTkxOTUtNDU1ZjgxOWRhYzFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI4VDA4NTU0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMyZDk2ZmZkYTE0ZWZmNjg0ZGZmODg4NjE3MjlkOTk0MTIxNTRjMjg0NmY2MTI0MDZiY2ZmZmI3MjQ4NTcxNmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BZhP32969copI5tkC7PtqCUSZBbarqDanuv0UOGtup0">

@omurozlu Manual process does the same comparison, while importing root and data disks of the instance. Is root disk of the instance, exists on that storage?

**Comment by nvazquez** [comment]
Hi @omurozlu - is the storage pool a datastore cluster type? Can you upload the management server logs for the import operation?

**Comment by omurozlu** [comment]
hello @sureshanaparti,
same zone added 
<img width="2269" alt="Screenshot 2024-02-28 at 15 35 46" src="https://github.com/apache/cloudstack/assets/41322804/a876bf5b-16c7-439a-8854-500185923566">

<img width="2277" alt="Screenshot 2024-02-28 at 15 36 02" src="https://github.com/apache/cloudstack/assets/41322804/da2e071e-e016-4392-8c5d-d1cc3ab8b710">

hello @nvazquez ,
datastore cluster type VMFS

management server log this : 
`2024-02-28 15:48:59,347 DEBUG [c.c.a.ApiServer] (qtp1278254413-352:ctx-5219e9e4 ctx-3aeefedf) (logid:15f58092) CIDRs from which account 'Account [{"accountName":"admin","id":2,"uuid":"683aa0c3-d606-11ee-8b6a-0050568d528d"}]' is allowed to perform API calls: 0.0.0.0/0,::/0
2024-02-28 15:48:59,349 INFO  [o.a.c.a.DynamicRoleBasedAPIAccessChecker] (qtp1278254413-352:ctx-5219e9e4 ctx-3aeefedf) (logid:15f58092) Account [Account [{"accountName":"admin","id":2,"uuid":"683aa0c3-d606-11ee-8b6a-0050568d528d"}]] is Root Admin or Domain Admin, all APIs are allowed.
2024-02-28 15:48:59,350 WARN  [o.a.c.a.ProjectRoleBasedApiAccessChecker] (qtp1278254413-352:ctx-5219e9e4 ctx-3aeefedf) (logid:15f58092) Project is null, ProjectRoleBasedApiAccessChecker only applies to projects, returning API [importUnmanagedInstance] for user [User {"username":"admin","uuid":"683bac0e-d606-11ee-8b6a-0050568d528d"}.] as allowed.
2024-02-28 15:48:59,351 DEBUG [o.a.c.a.StaticRoleBasedAPIAccessChecker] (qtp1278254413-352:ctx-5219e9e4 ctx-3aeefedf) (logid:15f58092) RoleService is enabled. We will use it instead of StaticRoleBasedAPIAccessChecker.
2024-02-28 15:48:59,351 DEBUG [o.a.c.r.ApiRateLimitServiceImpl] (qtp1278254413-352:ctx-5219e9e4 ctx-3aeefedf) (logid:15f58092) API rate limiting is disabled. We will not use ApiRateLimitService.
2024-02-28 15:48:59,353 WARN  [c.c.a.d.ParamGenericValidationWorker] (qtp1278254413-352:ctx-5219e9e4 ctx-3aeefedf) (logid:15f58092) Received unknown parameters for command importUnmanagedInstance. Unknown parameters : temppath importsource hypervisor host zoneid diskpath
2024-02-28 15:48:59,366 INFO  [o.a.c.f.j.i.AsyncJobMonitor] (API-Job-Executor-10:ctx-aa2475ff job-142) (logid:4377266b) Add job-142 into job monitoring
2024-02-28 15:48:59,374 DEBUG [o.a.c.f.j.i.AsyncJobManagerImpl] (qtp1278254413-352:ctx-5219e9e4 ctx-3aeefedf) (logid:15f58092) submit async job-142, details: AsyncJobVO: {id:142, userId: 2, accountId: 2, instanceType: None, instanceId: null, cmd: org.apache.cloudstack.api.command.admin.vm.ImportUnmanagedInstanceCmd, cmdInfo: {"temppath":"","migrateallowed":"true","httpmethod":"GET","ctxAccountId":"2","details[0].memory":"32768","cmdEventType":"VM.IMPORT","hypervisor":"vmware","host":"","ctxStartEventId":"373","ctxDetails":"{\"interface com.cloud.offering.ServiceOffering\":\"04a0bc45-5669-4ef7-8214-ea7bf8e4e87f\",\"interface com.cloud.org.Cluster\":\"07cc5a91-8306-4ae9-9a49-8b89965ea23f\"}","details[0].cpuNumber":"8","details[0].cpuSpeed":"1000","clusterid":"07cc5a91-8306-4ae9-9a49-8b89965ea23f","importsource":"unmanaged","nicnetworklist[0].network":"871e547e-d26a-43c0-9d7b-0c1f33a3edb8","serviceofferingid":"04a0bc45-5669-4ef7-8214-ea7bf8e4e87f","response":"json","ctxUserId":"2","displayname":"testvm","name":"WinTestYeni","zoneid":"3e34a0b7-ba5e-44f1-ab7a-8598549095dc","diskpath":"","nicnetworklist[0].nic":"Network adapter 1","username":""}, cmdVersion: 0, status: IN_PROGRESS, processStatus: 0, resultCode: 0, result: null, initMsid: 3232277003, completeMsid: null, lastUpdated: null, lastPolled: null, created: null, removed: null}
2024-02-28 15:48:59,374 DEBUG [o.a.c.f.j.i.AsyncJobManagerImpl] (API-Job-Executor-10:ctx-aa2475ff job-142) (logid:36097674) Executing AsyncJobVO: {id:142, userId: 2, accountId: 2, instanceType: None, instanceId: null, cmd: org.apache.cloudstack.api.command.admin.vm.ImportUnmanagedInstanceCmd, cmdInfo: {"temppath":"","migrateallowed":"true","httpmethod":"GET","ctxAccountId":"2","details[0].memory":"32768","cmdEventType":"VM.IMPORT","hypervisor":"vmware","host":"","ctxStartEventId":"373","ctxDetails":"{\"interface com.cloud.offering.ServiceOffering\":\"04a0bc45-5669-4ef7-8214-ea7bf8e4e87f\",\"interface com.cloud.org.Cluster\":\"07cc5a91-8306-4ae9-9a49-8b89965ea23f\"}","details[0].cpuNumber":"8","details[0].cpuSpeed":"1000","clusterid":"07cc5a91-8306-4ae9-9a49-8b89965ea23f","importsource":"unmanaged","nicnetworklist[0].network":"871e547e-d26a-43c0-9d7b-0c1f33a3edb8","serviceofferingid":"04a0bc45-5669-4ef7-8214-ea7bf8e4e87f","response":"json","ctxUserId":"2","displayname":"testvm","name":"WinTestYeni","zoneid":"3e34a0b7-ba5e-44f1-ab7a-8598549095dc","diskpath":"","nicnetworklist[0].nic":"Network adapter 1","username":""}, cmdVersion: 0, status: IN_PROGRESS, processStatus: 0, resultCode: 0, result: null, initMsid: 3232277003, completeMsid: null, lastUpdated: null, lastPolled: null, created: null, removed: null}
2024-02-28 15:48:59,375 DEBUG [c.c.a.ApiServlet] (qtp1278254413-352:ctx-5219e9e4 ctx-3aeefedf) (logid:15f58092) ===END===  192.168.161.8 -- GET  name=WinTestYeni&clusterid=07cc5a91-8306-4ae9-9a49-8b89965ea23f&displayname=testvm&zoneid=3e34a0b7-ba5e-44f1-ab7a-8598549095dc&importsource=unmanaged&hypervisor=vmware&host=&username=&diskpath=&temppath=&serviceofferingid=04a0bc45-5669-4ef7-8214-ea7bf8e4e87f&details[0].cpuNumber=8&details[0].cpuSpeed=1000&details[0].memory=32768&migrateallowed=true&nicnetworklist[0].nic=Network+adapter+1&nicnetworklist[0].network=871e547e-d26a-43c0-9d7b-0c1f33a3edb8&command=importUnmanagedInstance&response=json
2024-02-28 15:48:59,378 WARN  [c.c.a.d.ParamGenericValidationWorker] (API-Job-Executor-10:ctx-aa2475ff job-142 ctx-8dae85a2) (logid:36097674) Received unknown parameters for command importUnmanagedInstance. Unknown parameters : temppath importsource hypervisor host zoneid diskpath
2024-02-28 15:48:59,386 DEBUG [c.c.u.AccountManagerImpl] (API-Job-Executor-10:ctx-aa2475ff job-142 ctx-8dae85a2) (logid:36097674) Access granted to Account [{"accountName":"admin","id":2,"uuid":"683aa0c3-d606-11ee-8b6a-0050568d528d"}] to Service offering {"id":37,"name":"Custom","uuid":"04a0bc45-5669-4ef7-8214-ea7bf8e4e87f"}. by AffinityGroupAccessChecker
2024-02-28 15:48:59,388 TRACE [o.a.c.a.c.a.v.ImportUnmanagedInstanceCmd] (API-Job-Executor-10:ctx-aa2475ff job-142 ctx-8dae85a2) (logid:36097674) nic, 'Network adapter 1', goes on net, '871e547e-d26a-43c0-9d7b-0c1f33a3edb8'
2024-02-28 15:48:59,397 DEBUG [c.c.a.ApiServlet] (qtp1278254413-364:ctx-71cf1a66) (logid:5b984267) ===START===  192.168.161.8 -- GET  jobId=36097674-1d79-401d-ba43-130fde33d3c4&command=queryAsyncJobResult&response=json
2024-02-28 15:48:59,397 DEBUG [c.c.a.ApiServlet] (qtp1278254413-364:ctx-71cf1a66) (logid:5b984267) Two factor authentication is already verified for the user 2, so skipping
2024-02-28 15:48:59,402 DEBUG [c.c.a.m.DirectAgentAttache] (DirectAgent-199:ctx-09d81f14) (logid:99c7322f) Seq 1-3508304109721617290: Executing request
2024-02-28 15:48:59,402 DEBUG [c.c.h.v.r.VmwareResource] (DirectAgent-199:ctx-09d81f14) (logid:36097674) Executing resource command GetUnmanagedInstancesCommand: [].
2024-02-28 15:48:59,403 DEBUG [c.c.a.ApiServer] (qtp1278254413-364:ctx-71cf1a66 ctx-f1129f61) (logid:5b984267) CIDRs from which account 'Account [{"accountName":"admin","id":2,"uuid":"683aa0c3-d606-11ee-8b6a-0050568d528d"}]' is allowed to perform API calls: 0.0.0.0/0,::/0
2024-02-28 15:48:59,405 INFO  [o.a.c.a.DynamicRoleBasedAPIAccessChecker] (qtp1278254413-364:ctx-71cf1a66 ctx-f1129f61) (logid:5b984267) Account [Account [{"accountName":"admin","id":2,"uuid":"683aa0c3-d606-11ee-8b6a-0050568d528d"}]] is Root Admin or Domain Admin, all APIs are allowed.
2024-02-28 15:48:59,407 WARN  [o.a.c.a.ProjectRoleBasedApiAccessChecker] (qtp1278254413-364:ctx-71cf1a66 ctx-f1129f61) (logid:5b984267) Project is null, ProjectRoleBasedApiAccessChecker only applies to projects, returning API [queryAsyncJobResult] for user [User {"username":"admin","uuid":"683bac0e-d606-11ee-8b6a-0050568d528d"}.] as allowed.
2024-02-28 15:48:59,407 DEBUG [o.a.c.a.StaticRoleBasedAPIAccessChecker] (qtp1278254413-364:ctx-71cf1a66 ctx-f1129f61) (logid:5b984267) RoleService is enabled. We will use it instead of StaticRoleBasedAPIAccessChecker.
2024-02-28 15:48:59,408 DEBUG [o.a.c.r.ApiRateLimitServiceImpl] (qtp1278254413-364:ctx-71cf1a66 ctx-f1129f61) (logid:5b984267) API rate limiting is disabled. We will not use ApiRateLimitService.
2024-02-28 15:48:59,418 DEBUG [c.c.a.ApiServlet] (qtp1278254413-364:ctx-71cf1a66 ctx-f1129f61) (logid:5b984267) ===END===  192.168.161.8 -- GET  jobId=36097674-1d79-401d-ba43-130fde33d3c4&command=queryAsyncJobResult&response=json
2024-02-28 15:48:59,419 DEBUG [c.c.h.v.m.HostMO] (DirectAgent-199:ctx-09d81f14 esx02.flexapp360.cloud, job-142, cmd: GetUnmanagedInstancesCommand) (logid:36097674) find VM hypervisor name: WinTestYeni on host
2024-02-28 15:48:59,419 INFO  [c.c.h.v.m.HostMO] (DirectAgent-199:ctx-09d81f14 esx02.flexapp360.cloud, job-142, cmd: GetUnmanagedInstancesCommand) (logid:36097674) VM hypervisor name: WinTestYeni not found in host cache
2024-02-28 15:48:59,419 DEBUG [c.c.h.v.m.HostMO] (DirectAgent-199:ctx-09d81f14 esx02.flexapp360.cloud, job-142, cmd: GetUnmanagedInstancesCommand) (logid:36097674) load VM cache on host
2024-02-28 15:48:59,459 DEBUG [c.c.a.ApiServlet] (qtp1278254413-352:ctx-944e5120) (logid:b2962708) ===START===  192.168.161.8 -- GET  clusterid=07cc5a91-8306-4ae9-9a49-8b89965ea23f&page=1&pagesize=10&command=listUnmanagedInstances&response=json
2024-02-28 15:48:59,459 DEBUG [c.c.a.ApiServlet] (qtp1278254413-352:ctx-944e5120) (logid:b2962708) Two factor authentication is already verified for the user 2, so skipping
2024-02-28 15:48:59,460 DEBUG [c.c.a.ApiServlet] (qtp1278254413-1586:ctx-b445e3bf) (logid:e3503aed) ===START===  192.168.161.8 -- GET  listall=true&clusterid=07cc5a91-8306-4ae9-9a49-8b89965ea23f&page=1&pagesize=10&command=listVirtualMachines&response=json
2024-02-28 15:48:59,460 DEBUG [c.c.a.ApiServlet] (qtp1278254413-1586:ctx-b445e3bf) (logid:e3503aed) Two factor authentication is already verified for the user 2, so skipping
2024-02-28 15:48:59,466 DEBUG [c.c.a.ApiServer] (qtp1278254413-352:ctx-944e5120 ctx-3e42fa19) (logid:b2962708) CIDRs from which account 'Account [{"accountName":"admin","id":2,"uuid":"683aa0c3-d606-11ee-8b6a-0050568d528d"}]' is allowed to perform API calls: 0.0.0.0/0,::/0
2024-02-28 15:48:59,466 DEBUG [c.c.a.ApiServer] (qtp1278254413-1586:ctx-b445e3bf ctx-74422909) (logid:e3503aed) CIDRs from which account 'Account [{"accountName":"admin","id":2,"uuid":"683aa0c3-d606-11ee-8b6a-0050568d528d"}]' is allowed to perform API calls: 0.0.0.0/0,::/0
2024-02-28 15:48:59,468 INFO  [o.a.c.a.DynamicRoleBasedAPIAccessChecker] (qtp1278254413-352:ctx-944e5120 ctx-3e42fa19) (logid:b2962708) Account [Account [{"accountName":"admin","id":2,"uuid":"683aa0c3-d606-11ee-8b6a-0050568d528d"}]] is Root Admin or Domain Admin, all APIs are allowed.
2024-02-28 15:48:59,468 INFO  [o.a.c.a.DynamicRoleBasedAPIAccessChecker] (qtp1278254413-1586:ctx-b445e3bf ctx-74422909) (logid:e3503aed) Account [Account [{"accountName":"admin","id":2,"uuid":"683aa0c3-d606-11ee-8b6a-0050568d528d"}]] is Root Admin or Domain Admin, all APIs are allowed.
2024-02-28 15:48:59,470 WARN  [o.a.c.a.ProjectRoleBasedApiAccessChecker] (qtp1278254413-1586:ctx-b445e3bf ctx-74422909) (logid:e3503aed) Project is null, ProjectRoleBasedApiAccessChecker only applies to projects, returning API [listVirtualMachines] for user [User {"username":"admin","uuid":"683bac0e-d606-11ee-8b6a-0050568d528d"}.] as allowed.
2024-02-28 15:48:59,470 WARN  [o.a.c.a.ProjectRoleBasedApiAccessChecker] (qtp1278254413-352:ctx-944e5120 ctx-3e42fa19) (logid:b2962708) Project is null, ProjectRoleBasedApiAccessChecker only applies to projects, returning API [listUnmanagedInstances] for user [User {"username":"admin","uuid":"683bac0e-d606-11ee-8b6a-0050568d528d"}.] as allowed.
`

**Comment by nvazquez** [comment]
Thanks @omurozlu - I have quickly tested imports on NFS cluster and zone wide storage and couldn't hit the issue. It seems related to datastore cluster type, will try to replicate it using datastore cluster

**Comment by omurozlu** [comment]
4.18 version working but 4.19 not working

**Comment by nvazquez** [comment]
Hi @omurozlu - I have tested 4.19.0 creating a VM with a ROOT volume on a datastore cluster, unmanaging it and import it back and it succeeds. 

I have checked your logs and the images you have shared and I think the problem is that the datastore cluster in which the ROOT volume of the VM is located, is not added to CloudStack.

From the cmk output, CloudStack is looking for datastore FA360-DS02-ESX:
````
"Storage pool for disk Hard disk 1(123-2000) with datastore: FA360-DS02-ESX not found in zone ID: 3e34a0b7-ba5e-44f1-ab7a-8598549095dc"
````

However, from the images above, I only see one datastore with name FA360-DS01-ESX.

Can you double check the names on vCenter and CloudStack? In case you have renamed the datastore on vCenter, you should rename it as well on CloudStack for them to match

**Comment by omurozlu** [comment]
Hello again @nvazquez ,
I now tested again (reset system) 
<img width="340" alt="1" src="https://github.com/apache/cloudstack/assets/41322804/094ea383-1dc2-42f3-8c4b-1553a536cc3d">
<img width="1109" alt="2" src="https://github.com/apache/cloudstack/assets/41322804/712807fd-ef85-4ff2-8221-f39b2914a70a">
<img width="1673" alt="3" src="https://github.com/apache/cloudstack/assets/41322804/deac14cb-4b49-4b6f-a4c5-f5bcd467b6da">
<img width="1671" alt="4" src="https://github.com/apache/cloudstack/assets/41322804/197f0a17-6da0-4161-94e9-1a815623b2a5">
<img width="1669" alt="5" src="https://github.com/apache/cloudstack/assets/41322804/24371191-634b-4db0-8a0b-1cb0e8809294">
this first tested

==============
and im again reset system 
<img width="1680" alt="11" src="https://github.com/apache/cloudstack/assets/41322804/2a0f045d-2bbc-4531-9b74-b32e3fef2dfe">
<img width="1680" alt="12" src="https://github.com/apache/cloudstack/assets/41322804/2636186f-c969-4269-ad7b-da14f0a230ed">
<img width="1680" alt="13" src="https://github.com/apache/cloudstack/assets/41322804/980d1b7c-1f20-4bc5-a59e-cc362f2159e9">

all time give same error

**Comment by nvazquez** [comment]
Sorry @omurozlu I wrongly tested datastore cluster pools, but I see you are using VMFS pools instead. Can you share what's the output of this SQL query?

````
select * from storage_pool;
````

**Comment by omurozlu** [comment]
@nvazquez,
<img width="473" alt="Screenshot 2024-03-04 at 21 16 05" src="https://github.com/apache/cloudstack/assets/41322804/649e79f7-a039-4e84-a805-b27079136593">
<img width="1680" alt="Screenshot 2024-03-04 at 21 15 55" src="https://github.com/apache/cloudstack/assets/41322804/bafb264f-715e-46ac-b4d1-c99b6768d27f">

**Comment by nvazquez** [comment]
Thanks @omurozlu can you please upload the management server logs around the whole import operation? I want to compare the received unmanaged instance disk's information to the information of the pool in the database

**Comment by omurozlu** [comment]
hello again @nvazquez,

im again re install and I showed you everything I did
[https://flexapp360.cloud/video.gif](url)
[management-server.log](https://github.com/apache/cloudstack/files/14530407/management-server.log)

**Comment by nvazquez** [comment]
Thanks @omurozlu - can you please test the fix https://github.com/apache/cloudstack/pull/8759?

**Comment by omurozlu** [comment]
Hello @nvazquez,

Im reinstall cloudstack and im used this repo
`[cloudstack]
name=cloudstack
baseurl=https://download.cloudstack.org/testing/nightly/2024-03-07/el8/4.19
enabled=1
gpgcheck=0`

=> CloudStack 4.19.1.0-SNAPSHOT

but now same error give me

**Comment by nvazquez** [comment]
Hi @omurozlu, this fix is not yet merged so the nightly version does not contain the fix

**Comment by omurozlu** [comment]
@nvazquez hello,
I have to wait for the version update ?

sorry i dont know how build project :)

**Comment by nvazquez** [comment]
@omurozlu I can give you the steps to build:

````
yum install git maven rpm-build (could be more required, can check executing the last step if it requires more packages)

git clone https://github.com/apache/cloudstack.git
git clone https://github.com/shapeblue/cloudstack-nonoss.git
cp cloudstack-nonoss/*.jar cloudstack/deps
cd cloudstack
git fetch origin
git fetch origin pull/8759/head:pr/8759
git checkout pr/8759
cd deps/
./install-non-oss.sh
cd ../packaging
./package.sh -d centos8
````

**Comment by omurozlu** [comment]
@nvazquez,

im step step make but give me error all time.
<img width="1374" alt="Screenshot 2024-03-10 at 01 42 17" src="https://github.com/apache/cloudstack/assets/41322804/03e35885-d1cb-4ab4-85f4-68bd81b64f04">

**Comment by nvazquez** [comment]
@omurozlu you can consider running these commands on a VM different than the management server and copying the built commands once they are finished. For the failure above I suspect you are missing the Java SDK 11, you can try installing openjdk-11

**Comment by nvazquez** [comment]
Btw I've updated the last command, it was missing `-p noredist`

**Comment by omurozlu** [comment]
Hello again @nvazquez,

im again trying,
1- ) install centos 7 and update
2-) install java-11-openjdk, java-11-openjdk-devel, gcc, glibc-devel, nodejs
3-) change java version
4-)
`yum install git maven rpm-build (could be more required, can check executing the last step if it requires more packages)

git clone https://github.com/apache/cloudstack.git
git clone https://github.com/shapeblue/cloudstack-nonoss.git
cp cloudstack-nonoss/*.jar cloudstack/deps
cd cloudstack
git fetch origin
git fetch origin pull/8759/head:pr/8759
git checkout pr/8759
cd deps/
./install-non-oss.sh
cd ../packaging
FLAGS=-DskipTests ./package.sh -p noredist -d centos8`

and still give me error 

<img width="1188" alt="Screenshot 2024-03-11 at 22 17 10" src="https://github.com/apache/cloudstack/assets/41322804/a8245cc8-ac27-443f-bdc8-dd5601fd1289">


Your information is very valuable, thank you

**Comment by nvazquez** [comment]
Hi @omurozlu the fix has been merged today and will be available on the 15/03 nightly packages on the 4.19 branch

**Comment by omurozlu** [comment]
hello @nvazquez 

> Hi @omurozlu the fix has been merged today and will be available on the 15/03 nightly packages on the 4.19 branch

im tested this version and working.

I have a few different questions
1-) When will this update be merged?
2-) Or how can I build the stable version + this pr ?

**Comment by nvazquez** [comment]
Thanks @omurozlu - I will close this issue as per your confirmation that is fixed.

The fix is already merged, it will be available on the next minor release for 4.19 which will be 4.19.1 and onwards. 4.19.1 can be expected on Q2 of this year (https://lists.apache.org/thread/0zg248b66fxtw1x99bmysjm1g7rygnmw).

To build 4.19.0 + this fix you can try the following:

````
git checkout 4.19.0.0
Modify the file changed on the PR: https://github.com/apache/cloudstack/pull/8759/files
cd packaging
FLAGS=-DskipTests ./package.sh -p noredist -d centos8
````
